pkg/apis/core/types_secretbinding.go

```diff
@@ -31,6 +31,17 @@ type SecretBinding struct {
 	SecretRef corev1.SecretReference
 	// Quotas is a list of references to Quota objects in the same or another namespace.
 	Quotas []corev1.ObjectReference
+	// Provider defines the provider type of the SecretBinding.
+	Provider *SecretBindingProvider
+}
+
+// SecretBindingProvider defines the provider type of the SecretBinding.
+type SecretBindingProvider struct {
+	// Type is the type of the provider.
+	//
```

Comment:
Nit: I think this empty line should rather not be there.
```suggestion
```

Reply:
I think the empty line actually help to separate the 2 logical things (the general field description and the note about the backwards compatibility note for multiple providers). But if you insist, I can remove it of course.

Reply:
I don't insist, and I agree in general it makes sense to have such a blank line. However it's quite inconsistent with descriptions of other fields, there are hardly any blank lines there at the moment. Also, there are no blank lines between the fields themselves, so such a blank line is rather conspicuous, for no good (or good enough) reason.